src/response/mod.rs

```diff
@@ -13,11 +11,28 @@ mod list_signing_keys_response;
 pub use self::cache_dictionary_fetch_response::*;
 pub use self::cache_dictionary_get_response::*;
 pub use self::cache_dictionary_increment_response::*;
-pub use self::cache_dictionary_set_response::*;
 pub use self::cache_get_response::*;
 pub use self::cache_set_fetch_response::*;
-pub use self::cache_set_response::*;
 pub use self::create_signing_key_response::*;
 pub use self::error::*;
 pub use self::list_cache_response::*;
 pub use self::list_signing_keys_response::*;
+
+macro_rules! empty_response {
+    ($name:ident) => {
+        /// Response type for a response with no data.
+        #[derive(Debug, Clone)]
+        pub struct $name(());
+
+        impl $name {
+            pub(crate) fn new() -> Self {
+                Self(())
+            }
+        }
+    };
+}
+
+empty_response!(MomentoSetResponse);
+empty_response!(MomentoDictionarySetResponse);
+empty_response!(MomentoDictionaryDeleteResponse);
+empty_response!(MomentoDeleteResponse);
```

Comment:
Instead of a new macro and a manually-defined default function, isn't is simpler to do

```rust
#[derive(Default)]
pub struct MomentoSetResponse {}

#[derive(Default)]
pub struct MomentoDeleteResponse {}

#[derive(Default)]
pub struct MomentoDictionarySetResponse {}

#[derive(Default)]
pub struct MomentoDictionaryDeleteResponse {}
```
?

Then below you do either `MomentoSetResponse::default()` or just `Default::default()`

Reply:
- I don't want to have them derive `Default` since there's no guarantee that new fields added in the future will support default. The `new` function is only visible within the crate so we can change it in the future.
- If we define them like that then users can do `MomentoSetResponse {}` and we won't be able to add new fields at all. (`#[non_exhaustive]` could solve this though)

Reply:
```rust
impl MomentoSetResponse { pub(crate) fn new() -> Self { Self {} } }
impl MomentoDeleteResponse { pub(crate) fn new() -> Self { Self {} } }
impl MomentoDictionarySetResponse { pub(crate) fn new() -> Self { Self {} } }
impl MomentoDictionaryDeleteResponse { pub(crate) fn new() -> Self { Self {} } }
```

is more in line with our simple caveperson style than a new macro =)

Reply:
I have gone and fixed everything up to use the simpler style :)